src/core/trtserver.cc

```diff
@@ -188,13 +189,37 @@ TrtServerProtobuf::Serialize(const char** base, size_t* byte_size) const
   *byte_size = serialized_.size();
 }
 
+//
+// TrtServerMetrics
+//
+// Implementation for TRTSERVER_Metrics.
+//
+class TrtServerMetrics {
+ public:
+  TrtServerMetrics() = default;
+  void Serialize(const char** base, size_t* byte_size);
+
+ private:
+  std::string serialized_;
+};
+
+void
+TrtServerMetrics::Serialize(const char** base, size_t* byte_size)
+{
+  serialized_ = ni::Metrics::SerializedMetrics();
```

Comment:
Wrap parts related to `ni::Metrics` with `#ifdef TRTIS_ENABLE_METRICS` and return error message otherwise?